app/controllers/sites_controller.rb

```diff
@@ -29,6 +29,7 @@ def update
 
   def build_site
     @site = @provider.sites.find { |site| site.id == params[:id] }
+    @site_name = @site.location_name.dup
```

Comment:
This threw me because I didn't understand that you were trying to keep the old location_name that precedes the update. Maybe `@previous_site_name` or `@site_name_before_update`?